v2/internal/frontend/desktop/common/uri_translate.go

```diff
@@ -1,20 +1,34 @@
 package common
 
-import "net/url"
+import (
+	"fmt"
+	"net/url"
+)
 
-func TranslateUriToFile(uri string, expectedScheme string, expectedHost string) (file string, match bool, err error) {
+var ErrUnexpectedScheme = fmt.Errorf("unexpected scheme")
+var ErrUnexpectedHost = fmt.Errorf("unexpected host")
```

Comment:
more of a stylistic suggestion coming from the rest of the code base
```
// ErrUnexpectedScheme some explanation, ie when does this error can arise.
// Link to github issue maybe?
const ErrUnexpectedScheme = "unexpected scheme"
// ErrUnexpectedHost same.
const ErrUnexpectedHost = "unexpected host"
```
and `fmt.Error(ErrUnexpectedScheme)` when returning.

Reply:
Shouldn't we try to keep well established patterns, especially patterns that are widely used in the go standard lib? Cause that's how gophers are used too and makes it easier to get familiar with the code base. Defining string constants and returning always a new error would make it impossible to use `errors.Is` or `errors.As` and especially it would also not be possible to check if the error has been wrapped and is somewhere in the error chain. Simple comparisons would need to be done by first getting the error string and comparing on that one. AFAIK I've never seen such a pattern in the go standard lib.
I've also seen that the PR isn't up to date with my latest version, I've replaced the error comparisons with `errors.Is` but haven't yet pushed it because I didn't know how @leaanthony wanted the proceed after his first comment.

As for adding comments I agree with you.

Reply:
@stffabi - push changes, all good. I think this is 99% there and ready to merge.